beacon-chain/slasher/receive.go

```diff
@@ -131,6 +135,10 @@ func (s *Service) processQueuedBlocks(ctx context.Context, epochTicker <-chan ty
 				continue
 			}
 			processedBlocksTotal.Add(float64(len(blocks)))
+
+			if err := s.serviceCfg.Database.PruneProposals(ctx, currentEpoch, s.params.historyLength); err != nil {
```

Comment:
If this is an expensive operation, we should look into putting it in another routine. Perhaps another function that runs on an epoch ticker and runs prune attestations and prune proposals